### DISCUSSION TITLE
A confusing piece of code in OutputCacheProvider

### DISCUSSION THREAD

**Discussion by thienvc** [dnnsoftware/Dnn.Platform#6258] [General]
https://github.com/dnnsoftware/Dnn.Platform/blob/3e727b08eae9ea5b05a675e987636afb2768c634/DNN%20Platform/HttpModules/OutputCaching/OutputCacheModule.cs#L168-L180

What does the `includeVaryByKeys `variable do anyway?
In the end, I found it useless
This could very well lead to a Cache Overflow error
Can someone who knows this place explain it to me?

**Comment by zyhfish** [comment]
this is used to avoid invalid cache content, if the page content is decided by some query string, for example, articleid, this can make sure the cache content will be based on article id and won't display error content to different article id values.

**Reply by thienvc** [reply]
I know, but try looking at the code to see where that variable is used?

**Reply by zyhfish** [reply]
https://github.com/dnnsoftware/Dnn.Platform/blob/3e727b08eae9ea5b05a675e987636afb2768c634/DNN%20Platform/HttpModules/OutputCaching/OutputCacheModule.cs#L202

**Reply by thienvc** [reply]
> https://github.com/dnnsoftware/Dnn.Platform/blob/3e727b08eae9ea5b05a675e987636afb2768c634/DNN%20Platform/HttpModules/OutputCaching/OutputCacheModule.cs#L202

Omg, im sorry
